### ISSUE TITLE
Type mismatch causes Addressable::URI#query_values= to fail on sort! with "comparison of Array with Array failed"

### ISSUE THREAD

**Issue by rehevkor5** [sporkmonger/addressable#94]
If the data passed to `query_values=` results in a type mismatch in the value of `new_query_values` on line 1504 when `.sort!` is called, such as:

`[['a', '1'], ['a', 2]]`

the method will fail with ``sort!': comparison of Array with Array failed`

I'd submit a PR fix myself, but I'm not sure what the desired solution is.

**Comment by dmitry** [comment]
Had the same issue, solved by `stringify_keys`.

**Comment by sporkmonger** [comment]
@dmitry It's come up before and as before, the answer is essentially no, won't fix. If Ruby adds an indifferent access hash structure to the standard library, I'll revisit that answer, but until then, I prefer for this kind of usage to be explicitly controlled by the user of the library, exactly as you're doing. I strongly argue that principle of least surprise was violated when Rails introduced this pattern in the first place and I reject the idea that once they made bad behavior commonplace that suddenly good behavior became "surprising".

**Comment by dmitry** [comment]
@sporkmonger I agree, but in that case the small note should be added to the documentation, as keys should be strings, and not symbols because of the ambiguous while sorting and forming the query. Somewhere here:

https://github.com/sporkmonger/addressable/blob/master/lib/addressable/uri.rb#L1607
